server/session/handler_anvil.go

```diff
@@ -0,0 +1,224 @@
+package session
+
+import (
+	"fmt"
+	"github.com/df-mc/dragonfly/server/block"
+	"github.com/df-mc/dragonfly/server/item"
+	"github.com/df-mc/dragonfly/server/world/sound"
+	"github.com/sandertv/gophertunnel/minecraft/protocol"
+	"math/rand"
+)
+
+// handleCraftRecipeOptional handles the CraftRecipeOptional request action, sent when taking a result from an anvil
+// menu. It also contains information such as the new name of the item and the multi-recipe network ID.
+func (h *ItemStackRequestHandler) handleCraftRecipeOptional(a *protocol.CraftRecipeOptionalStackRequestAction, s *Session, filterStrings []string) error {
+	// First check if there actually is an anvil opened.
+	if !s.containerOpened.Load() {
+		return fmt.Errorf("no anvil container opened")
+	}
+
+	w := s.c.World()
+	pos := s.openedPos.Load()
+	anvil, ok := w.Block(pos).(block.Anvil)
+	if !ok {
+		return fmt.Errorf("no anvil container opened")
+	}
+	if len(filterStrings) < int(a.FilterStringIndex) {
+		return fmt.Errorf("filter string index %v is out of bounds", a.FilterStringIndex)
+	}
+
+	first, _ := h.itemInSlot(protocol.StackRequestSlotInfo{
+		ContainerID: containerAnvilInput,
+		Slot:        1,
+	}, s)
+	if first.Empty() {
+		return fmt.Errorf("no item in first input slot")
+	}
+	result := first
+
+	second, _ := h.itemInSlot(protocol.StackRequestSlotInfo{
+		ContainerID: containerAnvilMaterial,
+		Slot:        0x2,
```

Comment:
You use a decimal `1` above, but a hexadecimal `0x2` here. Might want to make those two consistent just for the sake of it.